### ISSUE TITLE
Updated API Types for TIU Interfaces

### ISSUE THREAD

**Issue by BerndHekele** [openETCS/modeling#83]
refers to https://github.com/openETCS/modeling/commit/3c97d040d4f791265073d51e56a6a29a233f9db3#commitcomment-9273585

Dear Partner,
I have made some changes in the API-Type Definitiuons for the Train Interfac es (TIU).
Especially, I have taken out the "info not available" default elemnt in all definitions.
In places where a not yet defined needs to be added to indicate a "unknown" state I will add a specific one on your feedback.

Please have a look at the deliverable:
@MariellePetitDoche @BaseliyosJacob @Farhangi @UweSteinkeFromSiemens @BenjaminBeichler @christianstahl .

**Comment by MERCEmentre** [comment]
Hello Bernd.

Where have you done such changes. I looked at TIU_Types_Pkg of the SCADE model but did not find such changes (e.g. for M10_brake_signal_command_T).

Best regards,
david

**Comment by BerndHekele** [comment]
Dear David,
I started to work with the interface and made some minor adjustments.
Updates are uploaded.
But, I also recognised some open issues:
1. M_train_data_entry_type_T	Question: Who needs it? Where does it come from?
2. There are some types incomplete defined. e.g., 
NID_ctraction_T	int	
nothing_to_resume_profile_follow_T
Passenger_door_control_info_T

Especially train_data_info_T looks like a gap and we need to close it.

Can you answer?

I try to make use of the comments in scade. Please, feel free to comment.

kr
Bernd




Can you make a statement on those?

**Comment by MariellePetitDoche** [comment]
Bernd,

I have a look according the information I need and I have detected some issues:
* cabActiveStatus_T is defined and use in the structure trainData_T. But trainData_T is never used in any structure. How can we have the information on which cabine is active ?
* isolationstatus :  this information is provided as output from EVC.  I need it inside the EVC, Which component provides this information ?
* same question how to have information on switch on or off of the power ?

**Comment by BerndHekele** [comment]
Dear Marielle,
 [ ] cabActiveStatus needs to be deleted. It is a part of the old model. You find the information now in the component:
trainstatus.m_cab_st of the new interface.  
 [ ] Isolation Status: topic is open and needs clarification. I assume the procedure handling Isolation is giving this infomation. But, it is not part of the current ianalysis.
 [ ] Power On/Off Status: topic is open and needs clarification. I would assume power-on/off is alway "on" of the own system since other wise the evc could not be in the current state.

cu
Bernd

**Comment by MariellePetitDoche** [comment]
Dear Bernd,

Thanks for the answers, we will adapt the input of my model latter.
For the Cab, I already use trainstatus.m_cab_st, but I need also know on which cab the OBU is installed : Cab A or Cab B : For some conditions of mode transition I need to know that desk in Cab A is open and that the OBU is installed in Cab A.

Do you know where is defined this information ? I guess it is an input of the system or a parameter.

**Comment by BerndHekele** [comment]
For each train / lokomotive this setting (which of the desks is cab_a and cab_b) is fixed. So I assume, the parameter must be part of the initialisation during system start.

**Comment by MERCEmentre** [comment]
Hello Bernd,

Sorry for the delay in my reply. I should be more reactive on openETCS this week.

Regarding M_train_data_entry_type_T, definition comes from SUBSET-094, §8.3.3.35, used in message TIU-3-I-1 defined in §8.3.2.12. I don't know if this information is needed or not. I simply transcribed SUBSET-094 info.

NID_ctraction_T is defined in SUBSET-026, §7.5.1.86.1. It is a value between 0 and 1023 to uniquely identify the traction system.

Regarding nothing_to_resume_profile_follow_T, it is used in the following structure:
https://github.com/openETCS/modeling/blob/master/API/description/asn1/train-interface.asn#L148
In the original structure, described in §8.3.2.15 in SUBSET-094, depending on value Q_TRACKINIT (defined in SUBSET-026 / §7.5.1.141) either a profile is sent (value 0) or initial state should be resumed (value 1). Structure nothing_to_resume_profile_follow_T corresponds to value 0 of Q_TRACKINIT.

Regarding Passenger_door_control_info_T, there is a whole message to define it: message num 51, TIU-5-O-2 (§8.3.2.17) in SUBSET-094. It looked to me that it was a message to allow or not opening of doors. I don't know if such information is sent by trackside. As the message is pretty complicated, I side-step its definition until further input.

Best regards,
david

**Comment by BaseliyosJacob** [comment]
TIU: 
Cab Status:
Signal Type: Boolean
Interface: 

TR_OBU_CabStatusA, 
TR_OBU_CabStatusB

Isolation:

Signal Type: Boolean

Interface: OBU_TR_Isolation_Status.

It is a Signal from the EVC to the Train. Since the Train needs the Status of the EVC to be informed that the EVC is in Isolation mode. In this case the train needs to be supervised by another ATP system. 

Power:

As Bernd already mentioned. The EVC is always under power up. The Status is in No Power and will change to Standy By after opening one of the Desk. But the power Main switch should be out of the ETCS focus.

**Comment by BaseliyosJacob** [comment]
Cab Staus:

If
TR_OBU_CabStatusA=0
TR_OBU_CabStatusB=0
No Cab activated

If
TR_OBU_CabStatusA=1
TR_OBU_CabStatusB=0
Cab A activated

If
TR_OBU_CabStatusA=0
TR_OBU_CabStatusB=1
Cab B activated

If
TR_OBU_CabStatusA=1
TR_OBU_CabStatusB=1
Invalid

**Comment by MariellePetitDoche** [comment]
Baseliyos, Cab status is fine for me but not sufficient, and the fact that desk is open is not to be confused with  the fact that the power is switch on:
The power can be switch on and the desk stay closed in this case the system shall switch to SL mode.


Besides, if the iinformation desk_A_is open, we have to be sure that the current system is linked to Cab A to switch to SB, if it is linked to Cab_B and the information cab_B is_open is not receive, the system can not switch to SB. Thus we need a parameter (or an input)  to identify on which cab is installed the software.

"But the power Main switch should be out of the ETCS focus." :+1: In this case you have to be clear on the management of the required operation in NP mode (cf Table 4.5.2)  in particular EB menagement and Determine Mode Management.

**Comment by BaseliyosJacob** [comment]
Baseliyos, Cab status is fine for me but not sufficient, and the fact that desk is open is not to be confused with the fact that the power is switch on:
The power can be switch on and the desk stay closed in this case the system shall switch to SL mode.

Yes, that’s clear for me. 

Besides, if the iinformation desk_A_is open, we have to be sure that the current system is linked to Cab A to switch to SB, if it is linked to Cab_B and the information cab_B is_open is not receive, the system can not switch to SB. Thus we need a parameter (or an input) to identify on which cab is installed the software.

O.k.!! Let me give  my experience understanding from the project.
1. If we talk about a Locomotive there is just one Cab – so no effect.
2. If we talk about a EMU (e.g. ICE, TGV, ..) there are 2 Cabines and 2 ETCS OBU. One OBU that will belongs to Cab A and the other one belongs to Cab B. 

"But the power Main switch should be out of the ETCS focus."  In this case you have to be clear on the management of the required operation in NP mode (cf Table 4.5.2) in particular EB menagement and Determine Mode Management.

Here i have to be more concrect. For my point of view the Train will go in a save status and break the power to the NP mode. In this case the driver has to reactivated the power switch. But the power switch should be general for the train Cab and in this case out of the scope of ETCS. From my point of view the EVC is in NP mode when the driver steps into the train, activated the cab main switch and turn the key. But may we should me more concret.

**Comment by JakobGartner** [comment]
I propose that for now we assume that the cab A/B is fixed/ known at start- up.

Our focus is to go through the scenarios as for our user stories on a reduced scenario for just 1 track.

Jakob

**Comment by BenjaminBeichler** [comment]
@BerndHekele 
Sorry for late answer: after digging into the definition I found many needed Data for SpeedSupervision, but for the Demonstration in March important Data definitions are not complete ... :-(

We talked long about the train data, which would be delivered by the TIU, as far as I see from the definition (see TIU_Types_Pkg::Message_Train_Interface_to_EVC_T.train_data_info). This Train Data should include A_break_emergency, Kdry_rst, Train length,  ... (take a look into §3.13.1.3 of the SRS). 

Some Stuff like brake_status is already defined, so I see this should be also the place for the other stuff.

@JakobGartner 
I think the Simulator is currently not build on this TIU definition, or at least will not deliver the values until the ITEA-review, I will use internal constants within the speed supervision to pretend an ICE 3 Train. After the Review we should refine the interface and should propagate this part of the API as storage for the Train Data.
